torch/nn/functional.py

```diff
@@ -3756,6 +3757,11 @@ def multi_head_attention_forward(query,                           # type: Tensor
     q = q * scaling
 
     if attn_mask is not None:
+
+        # convert ByteTensor attn_mask to float
+        if attn_mask.dtype == torch.bool:
+            attn_mask = torch.zeros(attn_mask.size()).masked_fill(attn_mask, float('-inf'))
```

Comment:
this will create `attn_mask` only on CPU. did you want to use sth similar to `torch.zeros_like(attn_mask, dtype = torch.float32)` semantics? Also masked_fill will create one more copy, should masked_fill_ be used instead? (if jit doesn't make any issue here)

Reply:
@vadimkantorov Thanks. Will make some changes.